File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int sum = 0; 
        int left = 0;
        int minLen = INT_MAX;
        for (int right = 0; right < nums.size(); right++) {
            // EXPAND
            sum += nums[right];
            // CHECK CONDITION TO REDUCE WINDOW SIZE
            while (sum >= target) {
                minLen = min(minLen, right - left + 1);
                sum -= nums[left];
                left++;
            }
        }
        return minLen == INT_MAX ? 0 : minLen;
    }
};

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        i, j = 0, 0
        SUM = 0

        ans = float('inf')

        while j < len(nums):
            SUM += nums[j]

            while SUM >= target :
                ans = min(ans, j - i + 1)

                SUM -= nums[i]
                i += 1
            
            j += 1
        
        return 0 if ans == float('inf') else ans